src/components/Todos/Todos.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  memo,
+  useRef,
+  useEffect,
+  useContext,
+} from 'react';
+import autoAnimate from '@formkit/auto-animate';
+import AddTodoForm from './AddTodoForm';
+import { TodosList } from './TodosList';
+import { updateTodo } from '../../api/todos';
+import { TodosError } from './TodosError';
+import { TodosFooter } from './TodosFooter';
+import { TodosContext } from './TodosContext';
+
+export const Todos = memo(
+  () => {
+    const { todos, setTodos } = useContext(TodosContext);
+    const parentRef = useRef(null);
+
+    useEffect(() => {
+      if (parentRef.current) {
+        autoAnimate(parentRef.current);
+      }
+    }, [parentRef]);
+
+    const checkAll = () => {
+      setTodos(prev => prev.map(todo => {
+        if (!todo.completed) {
+          updateTodo(todo.id, { completed: true });
+        }
+
+        return {
+          ...todo,
+          completed: true,
+        };
+      }));
+    };
+
+    const allChecked = todos.every(todo => todo.completed);
+    const todosCount = todos.length;
+
+    return (
+      <div className="todoapp">
+        <h1 className="todoapp__title">todos</h1>
+        <div ref={parentRef} className="todoapp__content">
+          <header className="todoapp__header">
+            <button
+              data-cy="ToggleAllButton"
+              type="button"
+              className={`todoapp__toggle-all ${allChecked ? 'active' : ''}`}
```

Comment:
classnames lib

Reply:
I thought this was pretty cool 😄 but as you wish

Reply:
hi @UmizDemud 
can you share what particular vulnerabilities prevent you from using the `classnames` package?
i'm not particularly bind to it, yet it's very widespread for a reason.